File: edentalbackend-main/DentalOffice.Dtos/PaymentDto.cs
using System.ComponentModel.DataAnnotations;

namespace DentalOffice.Dtos
{
    public record PaymentDto
    {
        public int Id { get; set; }
        [Required]
        public DateTime Date { get; set; }
        [Required]
        public decimal Amount { get; set; }
        [Required]
        public int UserId { get; set; }
        [Required]
        public int TreatmentId { get; set; }
        public string CardNumber { get; set; }
        public string PaymentIntent { get; set; }
        public string Client { get; set; }
        public string TreatmentName { get; set; }
        public string DentistName { get; set; }
        public int DentistId { get; set; }
    }
}

File: eDental-flutter-main/lib/apimodels/paymentDto.dart
import 'package:dart_json_mapper/dart_json_mapper.dart';

import '../helpers/dateJsonFormatter.dart';

@jsonSerializable
class PaymentDto {
  int id;
  @JsonProperty(converter: CustomDateTimeConverter())
  DateTime date;
  double amount;
  int userId;
  int treatmentId;
  String cardNumber;
  String client;
  String treatmentName;
  String? paymentIntent;

  PaymentDto(this.id, this.date, this.amount, this.userId, this.treatmentId,
      this.cardNumber, this.client, this.treatmentName, this.paymentIntent);
}
